XMonad/Actions/WorkspaceNames.hs

```diff
@@ -185,3 +196,21 @@ workspaceNamePrompt conf job = do
                                 Just i -> i
         contains completions input =
           return $ filter (Data.List.isInfixOf input) completions
+
+setTag :: (WorkspaceId -> WorkspaceId) -> WindowSpace -> WindowSpace
+setTag remap ws = ws { W.tag = remap $ W.tag ws }
+
+ewmhWorkspaceNamesLogHook' :: ((WorkspaceId -> Maybe String) -> WorkspaceId -> String)
+                           -> X ()
+ewmhWorkspaceNamesLogHook' nameWorkspace =
+  map . setTag . nameWorkspace <$> getWorkspaceNames' >>=
+      ewmhDesktopsLogHookCustom
+
+ewmhWorkspaceNamesLogHook = ewmhWorkspaceNamesLogHook' getWorkspaceNameFromTag
+
+ewmhWorkspaceNames :: XConfig a -> XConfig a
+ewmhWorkspaceNames c = c { startupHook     = startupHook c +++ ewmhDesktopsStartup
+                         , handleEventHook = handleEventHook c +++ ewmhDesktopsEventHook
+                         , logHook         = logHook c +++ ewmhWorkspaceNamesLogHook
+                         }
+  where x +++ y = mappend y x
```

Comment:
> The right way to do this is probably to provide an appropriate function for use with ewmhDesktopsLogHookCustom.

This behavior can't be accomplished with a pure function because access to X is needed.

> I don't understand why there is EWMH code in WorkspaceNames.hs. Anything with ewmh in the name should be moved out of this file and into EwmhDesktops.hs.

The behavior of these functions is specific to WorkspaceNames which is functionality that I imagine most people don't use. It seemed more appropriate to me to group this with WorkspaceNames than with ewmh.

> Also, your ewmhWorkspaceNames function is almost an exact copy of the ewmh in EwmhDesktops.hs.

Except that it uses the workspace names log hook instead of the one that just uses workspace tags. If I don't add this function, then users would have to manually set the startup, handleEvent and log hooks.

Is your argument that you can do something like

```
(ewmh def) {
   logHook = (logHook def) +++ ewmhWorkspaceNamesLogHook
}
```

to acheive the same effect?